cfn-resources/auditing/mongodb-atlas-auditing.json

```diff
@@ -40,7 +40,8 @@
     "/properties/AuditAuthorizationSuccess"
   ],
   "required": [
-    "ProjectId"
+    "ProjectId",
+    "AuditFilter"
```

Comment:
Was curious about this change in the AuditFilter field being required. I can see that in terraform it is also defined as optional https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_auditing.go#L42, and did a small test run without defining the audit_filter attribute and creation was successful. Let me know if you can verify the same, and would be worth verifying with upstream team if annotation in mms is accurate before we do this breaking change (also needed in terraform if applicable).

Reply:
@AgustinBettati We can sync on this because for me not setting auditFilter did not work and I kept on getting an error for invalid auditFilter and that's how I discovered this issue. Also referring API doc for this endpoint [here](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Auditing/operation/updateAuditingConfiguration)

Reply:
ok so after looking more into this, the attribute is marked as required in documentation and in the API but yes it does work without auditFilter if I make the call directly using Postman, I guess which is why the SDK takes it to be a string and not *string. Because of that I believe the SDK defaults it to `""` (haven't yet been able to log the exact API call yet) which breaks the API call and returns an error for invalid auditFilter.

Thanks @AgustinBettati for sharing this!

Reply:
Verified that in Terraform (using manual Atlas SDK) this attribute is not included in API request (for empty value ``json:"auditFilter,omitempty"`` takes care of it and the API call still works:
```
2023/09/22 15:28:49 [DEBUG] MongoDB Atlas API Request Details:
---[ REQUEST ]---------------------------------------
PATCH /api/atlas/v1.0/groups/....../auditLog HTTP/1.1
{
 "auditAuthorizationSuccess": true,
 "enabled": true
}

-----------------------------------------------------
2023/09/22 15:28:49 [DEBUG] MongoDB Atlas API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 202 Accepted
```

Reply:
Request Response from new SDK if auditFilter is not set:
```
PATCH /api/atlas/v2/groups/..../auditLog HTTP/1.1
Accept: application/vnd.atlas.2023-01-01+json
Content-Type: application/vnd.atlas.2023-01-01+json

{"auditAuthorizationSuccess":false,"auditFilter":"","enabled":true}


HTTP/2.0 400 Bad Request
Content-Type: application/vnd.atlas.2023-01-01+json;charset=utf-8
{"detail":"Invalid attribute auditFilter specified.","error":400,"errorCode":"INVALID_ATTRIBUTE","parameters":["auditFilter"],"reason":"Bad Request"}
```

Reply:
Nice work obtaining this detail, confirmed we will have to adjust from mms side side then

Reply:
yes, I will raise PR to update spec in mms so it can be made optional, we'll then need to use new version of the updated SDK to proceed on this in CFN

Reply:
updated